zp-relayer/workers/poolTxWorker.ts

```diff
@@ -63,64 +64,75 @@ export async function createPoolTxWorker<T extends EstimationType>(
         to: config.poolAddress,
         chainId: CHAIN_ID,
       }
+      const gasPriceValue = await gasPrice.fetchOnce()
+      const gasPriceWithExtra = addExtraGasPrice(gasPriceValue, config.gasPriceSurplus)
+      const [txHash, rawTransaction] = await signTransaction(
+        web3,
+        {
+          ...txConfig,
+          ...gasPriceWithExtra,
+        },
+        config.relayerPrivateKey
+      )
       try {
-        const gasPriceValue = await gasPrice.fetchOnce()
-        const gasPriceWithExtra = addExtraGasPrice(gasPriceValue, config.gasPriceSurplus)
-        const [txHash, rawTransaction] = await signTransaction(
-          web3,
-          {
-            ...txConfig,
-            ...gasPriceWithExtra,
-          },
-          config.relayerPrivateKey
-        )
         await sendTransaction(web3Redundant, rawTransaction)
-
-        await updateNonce(++nonce)
-
-        logger.info(`${logPrefix} TX hash ${txHash}`)
-
-        await updateField(RelayerKeys.TRANSFER_NUM, commitIndex * OUTPLUSONE)
-
-        const nullifier = getTxProofField(txProof, 'nullifier')
-        const outCommit = getTxProofField(txProof, 'out_commit')
-
-        const truncatedMemo = truncateMemoTxPrefix(rawMemo, txType)
-        const prefixedMemo = buildPrefixedMemo(outCommit, txHash, truncatedMemo)
-
-        pool.optimisticState.updateState(commitIndex, outCommit, prefixedMemo)
-        logger.debug('Adding nullifier %s to OS', nullifier)
-        await pool.optimisticState.nullifiers.add([nullifier])
-        const poolIndex = (commitIndex + 1) * OUTPLUSONE
-        logger.debug('Adding root %s at %s to OS', rootAfter, poolIndex)
-        await pool.optimisticState.roots.add({
-          [poolIndex]: rootAfter,
-        })
-
-        const sentJob = await sentTxQueue.add(
-          txHash,
-          {
-            poolJobId: job.id as string,
-            root: rootAfter,
-            outCommit,
-            commitIndex,
-            truncatedMemo,
-            nullifier,
-            txConfig,
-            txPayload: tx,
-            prevAttempts: [[txHash, gasPriceWithExtra]],
-          },
-          {
-            delay: config.sentTxDelay,
-            priority: nonce,
-          }
-        )
-
-        txHashes.push([txHash, sentJob.id as string])
       } catch (e) {
-        logger.error(`${logPrefix} Send TX failed: ${e}`)
```

Comment:
Should we still log an error, if it's different from insufficient balance one?

Reply:
In this case error will be logged by `withErrorLog` wrapper
https://github.com/zkBob/zeropool-relayer/blob/b90be9aa5497c52de561ce9e70e57ac3f90c1b5b/zp-relayer/workers/poolTxWorker.ts#L131